### ISSUE TITLE
Create Wrapper Class around Alsa and Effects Manager to Handle Threads and Blocking

### ISSUE THREAD

**Issue by iamdario** [Jscott44/RTEP5-the-raspberry-jam#20]
We will need to implement 3 threads:
- Capture
- Process
- Playback

While capture will automatically be blocked using Alsa's read function (which will unblock once the buffer is full and has been read), we require a way to block the process and playback threads until the read function has returned for the process thread and until the data has been processed for the playback thread. This wrapper class will use mutexes and conditional variables to block the threads until needed.

**Comment by iamdario** [comment]
This issue has been nullified. As mentioned in #23, we do not want to create a wrapper. Instead, callbacks between separate objects should be registered with each object responsible for their thread (rather than the wrapper being responsible for this). We will continue with the callback method instead.

**Comment by iamdario** [comment]
Has been merged into #18 in pull requests #24 and #25.
